Source/Particles/PhysicalParticleContainer.cpp

```diff
@@ -3705,21 +3725,175 @@ PhysicalParticleContainer::AccumulateVelocitiesAndComputeTemperature (
+        amrex::MultiFab*  w2y_mf   = local_temperature_arrays->get("w2", Direction{1}, lev);
+        amrex::MultiFab*  w2z_mf   = local_temperature_arrays->get("w2", Direction{2}, lev);
+        amrex::MultiFab*  vbarx_mf = local_temperature_arrays->get("vbar", Direction{0}, lev);
+        amrex::MultiFab*  vbary_mf = local_temperature_arrays->get("vbar", Direction{1}, lev);
+        amrex::MultiFab*  vbarz_mf = local_temperature_arrays->get("vbar", Direction{2}, lev);
+
+        // Normalize variance after accumulating sums cell by cell
+#ifdef AMREX_USE_OMP
+#pragma omp parallel if (amrex::Gpu::notInLaunchRegion())
 #endif
+        for ( amrex::MFIter mfi(*T_vf[lev][0], amrex::TilingIfNotGPU()); mfi.isValid(); ++mfi ) {
+
+            amrex::Array4<amrex::Real> const& varx_arr = T_vf[lev][0]->array(mfi);
+            amrex::Array4<amrex::Real> const& vary_arr = T_vf[lev][1]->array(mfi);
+            amrex::Array4<amrex::Real> const& varz_arr = T_vf[lev][2]->array(mfi);
+            const amrex::Array4<const int> & nx_arr = nx_mf->const_array(mfi);
+            const amrex::Array4<const int> & ny_arr = ny_mf->const_array(mfi);
+            const amrex::Array4<const int> & nz_arr = nz_mf->const_array(mfi);
+            const amrex::Array4<const amrex::Real> & wx_arr = wx_mf->const_array(mfi);
+            const amrex::Array4<const amrex::Real> & wy_arr = wy_mf->const_array(mfi);
+            const amrex::Array4<const amrex::Real> & wz_arr = wz_mf->const_array(mfi);
+            const amrex::Array4<const amrex::Real> & w2x_arr = w2x_mf->const_array(mfi);
+            const amrex::Array4<const amrex::Real> & w2y_arr = w2y_mf->const_array(mfi);
+            const amrex::Array4<const amrex::Real> & w2z_arr = w2z_mf->const_array(mfi);
+            amrex::Array4<amrex::Real> const& vxbar_arr = vbarx_mf->array(mfi);
+            amrex::Array4<amrex::Real> const& vybar_arr = vbary_mf->array(mfi);
+            amrex::Array4<amrex::Real> const& vzbar_arr = vbarz_mf->array(mfi);
+
+            const amrex::Box& tbx  = mfi.growntilebox( T_vf[lev][0]->ixType().toIntVect() );
+            const amrex::Box& tby  = mfi.growntilebox( T_vf[lev][1]->ixType().toIntVect() );
+            const amrex::Box& tbz  = mfi.growntilebox( T_vf[lev][2]->ixType().toIntVect() );
+
+
+            bool single_pass = (depos_type == warpx::particles::deposition::TemperatureDepositionType::SINGLE_PASS);
+
+            // Update Mean and Variance values after running through weight deposition loop
+            amrex::ParallelFor(tbx, tby, tbz,
+                [=] AMREX_GPU_DEVICE (int i, int j, int k) {
+                    if (nx_arr(i,j,k) > 1) {
+                        amrex::Real sumw = wx_arr(i,j,k);
+                        amrex::Real sumwv = vxbar_arr(i,j,k);
+                        amrex::Real n = static_cast<amrex::Real>(nx_arr(i,j,k));
+                        amrex::Real norm = n/((n-1._rt)*sumw);
```

Comment:
Why do you use the `n/(n-1)` factor?

Reply:
I am conforming to the algorithm in https://dl.acm.org/doi/pdf/10.1145/359146.359153 which uses sample variance normalization 1/(n-1) or Bessel's correction for normalization instead of population variance 1/n. I initially was using population variance, but transitioned it to the sample variance to stick to the algorithm in Bell. I think in terms of how the particles represent samples of real particles it likely makes sense to use.

Reply:
I guess the point is that the macro-particle population is intrinsically a sampled dataset. I have waffled back and forth on it a bit, but I think based on the thought experiment of how we load particles, by sampling a larger population, I would argue that we are not sampling the entire real population and we should have the correction included. I have plotted out both cases for each temperature component and computed the average temperatures of the corresponding temperature calculations after 1 small timestep. They particles are loaded in with 10 eV in each component. The sample variance calculation gets closer to the 10 eV that is expected.

![image](https://github.com/user-attachments/assets/20b157da-8ac2-4e9e-963f-dcce5cdb7970)